File: utils/has_columns_test.go
package utils

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestHasColumns(t *testing.T) {
	headerColumn := []string{"Item", "Description", "Amount"}
	requiredColumns := []string{"Amount", "Item"}

	assert.True(t, HasColumns(headerColumn, requiredColumns))
}

func TestDoesNotHaveColumns(t *testing.T) {
	headerColumn := []string{"Item", "Description", "Amount"}
	requiredColumns := []string{"Supermarket", "Item"}

	assert.False(t, HasColumns(headerColumn, requiredColumns)) // Corrected to assert.False
}

File: utils/hasColumns.test.ts
import { expect } from "@std/expect";
import { describe, it } from "@std/testing/bdd";
import { hasColumns } from "./hasColumns.ts";

describe("hasColumns", () => {
  it("positive case", () => {
    const headerColumn = ["Item", "Description", "Amount"];
    const requiredColumns = ["Amount", "Item"];
    expect(hasColumns(headerColumn, requiredColumns)).toBe(true);
  });
});

Deno.test("negative case", () => {
  const headerColumn = ["Item", "Description", "Amount"];
  const requiredColumns = ["Supermarket", "Item"];
  expect(hasColumns(headerColumn, requiredColumns)).toBe(false);
});
